dev/format.sh

```diff
@@ -1,46 +1,46 @@
 #! /usr/bin/sh
 
 format() {
-# don't use a directory with whitespace
-local whitelist="2d cuda gpu ml simulation stereo tracking"
-
-local guess_PCL_DIR="$(dirname "$0")/../"
-local guess_PCL_DIR=$(cd "${guess_PCL_DIR}"; pwd)
-local PCL_DIR="${PCL_DIR:-${guess_PCL_DIR}}"
-echo $PCL_DIR
-
-local formatter_versions="9 9.0 8 8.0 7 7.0 6 6.0"
-for version in ${formatter_versions}; do
-    local guess_formatter="${guess_formatter:-"$(which clang-format-${version})"}"
-done
-local guess_formatter="${guess_formatter:-"$(which clang-format)"}"
-local formatter="${1:-${guess_formatter}}"
-
-if [ ! -f "${formatter}" ]; then
-    echo "Could not find a clang-format. Please specify one as the first argument"
-    exit 166
-fi
-
-local version_string="$("${formatter}" -version | sed "s/^.*clang-format version \([0-9.-]*\).*$/\1/")"
-
-if [ "${version_string}" \< "6" ]; then
-    echo "Required version: >6.0.0; Found version: ${version_string}"
-    exit 167
-fi
-exit
-
-for dir in ${whitelist}; do
-    path=${PCL_DIR}/${dir}
-    if [ ! -d "${path}" ]; then
-        # This should not happen. guess_PCL_DIR should be sufficient in most cases
-        echo "Could not find ${dir} in ${PCL_DIR}. Is the PCL_DIR env var correct?"
-        exit 168
+    # don't use a directory with whitespace
+    local whitelist="2d cuda gpu ml simulation stereo tracking"
+
+    local guess_PCL_DIR="$(dirname "$0")/../"
```

Comment:
What about passing `PCL_DIR` as an argument to the script? Then no need to cd/pwd and guessing.

Reply:
2 separate reasons:
* reducing confusion and checking for mixing `clang-format-path` and `pcl-source-path`
* technically, the guess is very good except a corner case (soft-links along the `dirname $0`)

Only power users need to provide the PCL_DIR as input args or env var, and I figured the env var would be less confusing.

The change to cli arg input is easy I had that before I moved to env var based backup